deisctl/backend/fleet/journal_test.go

```diff
@@ -0,0 +1,75 @@
+package fleet
+
+import (
+	"bytes"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/coreos/fleet/machine"
+	"github.com/coreos/fleet/schema"
+)
+
+type mockJournalCommandRunner struct {
+	validUnits []string
+}
+
+func (mockJournalCommandRunner) LocalCommand(string) (int, error) {
+	return 0, nil
+}
+
+func (m mockJournalCommandRunner) RemoteCommand(cmd string, addr string, timeout time.Duration) (int, error) {
+	if addr != "1.1.1.1" || timeout != 0 {
+		return -1, fmt.Errorf("Got %s %d, which is unexpected", cmd, addr, timeout)
+	}
+
+	for _, unit := range m.validUnits {
+		if fmt.Sprintf("journalctl --unit %s --no-pager -n 40 -f", unit) == cmd {
+			return 0, nil
+		}
+	}
+
+	return -1, fmt.Errorf("Didn't find command %s to match with units %v", cmd, m.validUnits)
+}
+
+func TestJournal(t *testing.T) {
+	testMachines := []machine.MachineState{
+		machine.MachineState{
+			ID:       "test-1",
+			PublicIP: "1.1.1.1",
+			Metadata: nil,
+			Version:  "",
+		},
+	}
+
+	testUnits := []*schema.Unit{
+		&schema.Unit{
+			Name:         "deis-router@1.service",
+			CurrentState: "loaded",
+			MachineID:    "test-1",
+		},
+		&schema.Unit{
+			Name:         "deis-router@2.service",
+			CurrentState: "loaded",
+			MachineID:    "test-1",
+		},
+	}
+
+	c := &FleetClient{Fleet: &stubFleetClient{testUnits: testUnits, testMachineStates: testMachines,
+		unitsMutex: &sync.Mutex{}}}
+
+	testWriter := bytes.Buffer{}
+
+	err := doJournal(c, "router", &testWriter, mockJournalCommandRunner{validUnits: []string{"deis-router@1.service", "deis-router@2.service"}})
```

Comment:
Would it make more sense to make the writer and the  as struct fields of FleetClient instead of part of the method signature? I can't help but feel that we could simplify the logic by introducing a `Runner` field which is of type `commandRunner` as well as an `Output` field of type `io.Writer`. That way your stubFleetClient could just specify the `Runner` as type `mockCommandRunner` and call `.Journal()` directly.